terminal.js

```diff
@@ -46,30 +42,42 @@ class FileSystem {
         return fullPath;
     }
 
-    readDir(path, root) {
+    readDir(path, root, options = {}) {
+        const exploreDir = (directory, result) => {
+            const path = this.getFileFullPath(directory);
+            result[path] = Object.keys(directory.children);
+
+            // If the "recursive" option is true, we have to
+            // explore the current directory's children.
+            if (options.recursive) {
+                Object.values(directory.children)
+                    .filter(child => child.type === FileTypes.Directory)
+                    .forEach(subDirectory => exploreDir(subDirectory, result));
+            }
+
+            return result;
+        };
+
         const dir = this.getFile(path, root);
-        return Object.keys(dir.children);
+        return exploreDir(dir, {});
     }
 
-    /**
-     * @TODO: handle mkdir foo/bar
-     */
     createDir(path, root) {
-        if (!root.children[path]) {
-            root.children[path] = new File(FileTypes.Directory, path);
-            root.children[path].parent = root;
-        } else {
-            console.log('Directory already exists');
-        }
+        this.createFile(FileTypes.Directory, path, root);
     }
 
-    /**
-     * @TODO: handle touch foo/bar.txt
-     */
-    createFile(path, root) {
-        if (!root.children[path]) {
-            root.children[path] = new File(FileTypes.RegularFile, path);
-            root.children[path].parent = root;
+    createRegularFile(path, root) {
+        this.createFile(FileTypes.RegularFile, path, root);
+    }
+
+    createFile(fileType, path, root) {
+        const fileName = pathHelper.basename(path);
+        const parentDirPath = pathHelper.dirname(path);
+        const parentDir = this.getFile(parentDirPath, root);
+
+        if (!parentDir.children[fileName]) {
+            parentDir.children[fileName] = new File(fileType, fileName);
+            parentDir.children[fileName].parent = parentDir;
         } else {
             console.log('File already exists');
```

Comment:
This smells more like a `thow error`. IMO If an error is founded, the bash should not continue executing code